// vietnam-qr-pay/dist/constants/bank-code.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BankCode = void 0;
var BankCode;
(function (BankCode) {
    BankCode["ABBANK"] = "ABB";
    BankCode["ACB"] = "ACB";
    BankCode["AGRIBANK"] = "AGRIBANK";
    BankCode["BAC_A_BANK"] = "BAB";
    BankCode["BAOVIET_BANK"] = "BAOVIETBANK";
    BankCode["BANVIET"] = "BVB";
    BankCode["BIDC"] = "BIDC";
    BankCode["BIDV"] = "BID";
    BankCode["CAKE"] = "CAKE";
    BankCode["CBBANK"] = "VNCB";
    BankCode["CIMB"] = "CIMB";
    BankCode["COOP_BANK"] = "COOPBANK";
    BankCode["DBS_BANK"] = "DBS";
    BankCode["DONG_A_BANK"] = "DONGABANK";
    BankCode["EXIMBANK"] = "EIB";
    BankCode["GPBANK"] = "GPBANK";
    BankCode["HDBANK"] = "HDB";
    BankCode["HONGLEONG_BANK"] = "HLB";
    BankCode["HSBC"] = "HSBC";
    BankCode["IBK_HCM"] = "IBKHCM";
    BankCode["IBK_HN"] = "IBKHN";
    BankCode["INDOVINA_BANK"] = "IVB";
    BankCode["KASIKORN_BANK"] = "KBANK";
    BankCode["KIENLONG_BANK"] = "KLB";
    BankCode["KOOKMIN_BANK_HCM"] = "KBHCM";
    BankCode["KOOKMIN_BANK_HN"] = "KBHN";
    BankCode["LIENVIETPOST_BANK"] = "LPB";
    BankCode["MBBANK"] = "MBB";
    BankCode["MSB"] = "MSB";
    BankCode["NAM_A_BANK"] = "NAB";
    BankCode["NCB"] = "NVB";
    BankCode["NONGHYUP_BANK_HN"] = "NONGHYUP";
    BankCode["OCB"] = "OCB";
    BankCode["OCEANBANK"] = "OCEANBANK";
    BankCode["PGBANK"] = "PGB";
    BankCode["PUBLIC_BANK"] = "PBVN";
    BankCode["PVCOM_BANK"] = "PVCOMBANK";
    BankCode["SACOMBANK"] = "STB";
    BankCode["SAIGONBANK"] = "SGB";
    BankCode["SCB"] = "SCB";
    BankCode["SEA_BANK"] = "SSB";
    BankCode["SHB"] = "SHB";
    BankCode["SHINHAN_BANK"] = "SVB";
    BankCode["STANDARD_CHARTERED_BANK"] = "SC";
    BankCode["TECHCOMBANK"] = "TCB";
    BankCode["TIMO"] = "TIMO";
    BankCode["TPBANK"] = "TPB";
    BankCode["UBANK"] = "UBANK";
    BankCode["UNITED_OVERSEAS_BANK"] = "UOB";
    BankCode["VIB"] = "VIB";
    BankCode["VIET_A_BANK"] = "VAB";
    BankCode["VIET_BANK"] = "VBB";
    BankCode["VIETCOMBANK"] = "VCB";
    BankCode["VIETINBANK"] = "CTG";
    BankCode["VPBANK"] = "VPB";
    BankCode["VRB"] = "VRB";
    BankCode["WOORI_BANK"] = "WRB";
})(BankCode = exports.BankCode || (exports.BankCode = {}));
